pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -633,6 +633,12 @@ func (p *provider) Validate(spec clusterv1alpha1.MachineSpec) error {
 		return fmt.Errorf("failed to validate instance profile: %v", err)
 	}
 
+	if config.IsSpotInstance != nil && *config.IsSpotInstance {
+		if config.SpotMaxPrice == nil {
```

Comment:
I'm not sure about this change. In general, I'd prefer if we allow a `nil` value here. Since by default if `SpotMaxPrice` is not specified the current on demand price would be used.

Reply:
AWS api doesn't allow it as you saw during the crd migration tests ;)

Reply:
It didn't allow this because `undefined` was being passed as a value from the front-end.

Reply:
yeah, what' the point of having nil SpotMaxPrice if the instance IsSpotInstance

Reply:
Exactly, the reason why you need to set this up is, the max price will be the same as the on-demand machine price which makes kinda no sense.